### DISCUSSION TITLE
Accessing alternate characters automatically in Indesign

### DISCUSSION THREAD

**Discussion by dpor** [rsms/inter#604] [Q&A] [answered]
Hello there.
Is there a way to set the alternate closed 'a' as default in Indesign?
Which option do I need to select in order to set is a default style?
Any help would be much appreciated.
Screen grabs attached of the 'a' options and the type options within Indesign.
Cheers
Dave

<img width="1480" alt="image" src="https://github.com/rsms/inter/assets/3615115/e0e6f78a-01aa-43ee-ac9e-21a9f3fa7afc">
<img width="436" alt="image" src="https://github.com/rsms/inter/assets/3615115/f6c3af79-d4c3-42ce-afd0-4bcaa63fde0c">
<img width="701" alt="image" src="https://github.com/rsms/inter/assets/3615115/e4997ae0-225c-4180-97fd-a5c37b75f3fe">

**Comment by acordy** [comment, marked as answer]
Hi,

Normally you should be able to set this as a default by turning on the relevant stylistic set (in your screenshot above this would be under _OpenType_ > _Stylistic_ Sets > _[Set 11]_). Set 11 appears to be the one that controls the "_a_" variant, as tested here [https://fontdrop.info/](https://fontdrop.info/), where if you turn on the switch "_cv11_" you'll see the "_a_" variant in the preview.

I say that normally you should be able to do this, since for some reason it doesn't actually work in InDesign right now. If you look under the Stylistic Sets menu, you'll see four named sets ("_Open digits_", "_Disambiguation_", "_Round quotes commas_", and "_Disambiguation (no slashed zero)_"), and then a series of unnamed ones in the form of "_[Set XYZ]_". The named one **do work**, but the unnamed ones (or at least set 11) do not.

Maybe it's as simple as InDesign needing them to be named to function properly, or maybe it's something else. Given that there are four that are already named, maybe the additional sets have just not been fully implemented yet...

In any case, I think you have a few options:

- You can file an issue and wait to see if it gets resolved;
- I believe InDesign supports some kind of scripting (I'm unfamiliar with this though), and you may be able to write or find some kind of script that replaces all the default "_a_" with the alternate glyph;
- You can probably use an OpenType feature freezer like [https://twardoch.github.io/fonttools-opentype-feature-freezer/](https://twardoch.github.io/fonttools-opentype-feature-freezer/) to "freeze" that stylistic set as default, and export a new version of the typeface to use.

Hope that helps!

**Reply by dpor** [reply]
Ah nice one. 
Thanks for the information. 
Definitely clears things up for me. 
I thought I was going a bit mad as I couldn't work it out in Indesign and in Figma the alternate style works fine.
Ill try and file an issue and see if it gets updated.
Thanks again - much appreciated.

**Comment by acordy** [comment]
Hi again,

You may have seen my comment here: [https://github.com/rsms/inter/issues/605](https://github.com/rsms/inter/issues/605), in the end this doesn't look like an issue with Inter, but rather with InDesign. It doesn't look like they've implemented access to _Character Variants_ in the UI the way they have for _Stylistic Sets_, which is a shame.

I just wanted to add, besides what I suggested earlier, that you may find this project useful: [https://github.com/samiartur/Character-Variant-Java-Script-for-Indesign](https://github.com/samiartur/Character-Variant-Java-Script-for-Indesign).

Basically they're two scripts which let you more easily apply a character variant to a selection, one with a specific "cvXX" hard-coded into it, and one with a dialog prompt where you can enter the two-digit character variant code (for example "11" in your case). I tested the one with the prompt and it works as expected. Still inconvenient since it only works on selections and not the whole document, but probably better than changing the glyphs one by one. I'm sure there's a way to adapt the script to work on the whole document, but I don't know enough to suggest how.

Hope that helps!

**Reply by dpor** [reply]
Ah great - thanks @acordy 
Again, much appreciated for the info and input.

We are working on a large Indesign template style/working file for a client so they will need to easily use paragraph styles when they add new content etc.
So unless its a universal automatic and easily applied setting I dont think it would work in this situation sadly.

But I would say that this is the first time I have ever experienced any alt-character access issues with paragraph style in Indesign.
Most alternate characters should be able to be accessible through style sets for this exact purpose.
But then perhaps Im old and Indesign is becoming and old and useless application. hahaha
The alternate universal style works perfectly in Figma.

**Reply by acordy** [reply]
I commented on the issue you filed [https://github.com/rsms/inter/issues/605](https://github.com/rsms/inter/issues/605) and I think it's better if we continue our discussion here (since I don't think there's an issue with Inter and that it should probably be closed). But others may come across this problem with InDesign so any solutions or workarounds we find might be useful.

I'll repeat a bit of what I said in the other post for context here:

> Most alternate characters should be able to be accessible through style sets for this exact purpose.

Stylistic sets are meant to swap out whole sets of characters, for instance with Inter `ss02` _Disambiguation_ swaps out a whole series of glyphs for ones that resolve visual ambiguities. Character variants are meant to just change one glyph (and accented versions) for another, like with `cv11` swapping out the two-storey "a" for the single-storey one.

Making a stylistic set for just one character would be an (at least semantic) misuse of the feature. Bear in mind also that I believe stylistic sets are limited to just 20 options, whereas character variants can number 99 (`cv01` to `cv99` I believe). So font designers have to be discriminating in what stylistic sets they include, or they'll run out of them!

The problem is just that InDesign has exposed a UI for modifying _Stylistic Sets_ but not _Character Variants_. It's dumb, but there you have it. If we should be filing an issue or feature request with anyone, it should be with Adobe and not Inter. But I wouldn't hold my breath waiting for it to be implemented.

So, on to your use case: Is there a need to have both versions of the "a" in the documents being produced for the client? If you're planning on always using just the single-storey "a", then I think your best solution would be to create and distribute a version of the fonts with that character variant "frozen" as the default. So you'd make a version of Inter called _Inter for Client XYZ_ and use that. You could then set up all your styles to use that typeface and no one would ever have to think about it again.

Would that work?

**Reply by dpor** [reply]
But how is the alternate 'a' different from the 'r' style set?
Isn't this exactly the same? Just swapping out one character for a whole style?

<img width="310" alt="image" src="https://github.com/rsms/inter/assets/3615115/58cac30b-a969-47ba-9e6f-cc041106fa31">

**Reply by dpor** [reply]
yeah - creating a custom font for the client would be a solution for sure but having this font easily accessible for all end users directly through Google fonts is one of the reasons for going with it to begin with. 
That way we don't have to distribute assets to the users. Keeps things more simple.

**Reply by dpor** [reply]
Also, thanks for the response.
I very much appreciate you taking the time to explain and help me understand the issue :)

**Reply by acordy** [reply]
The alternate "a" isn't really a different case than the "r", but my impression is that having a stylistic set just for the curved "r" was also kind of a misuse of the feature, which is why it's no longer present in version 4 of Inter (currently in beta). That stylistic set exists in version 3 which is what you're seeing, but is no longer there in the upcoming version. I'm just guessing here, since I'm not the one working on Inter, but I believe that a single character variant doesn't warrant a whole stylistic set, which is why it's gone now. Again though, that's just my guess.

**Reply by dpor** [reply]
oh damn. I thought if the alternate 'r' style set existing maybe we could get the alternate 'a' style set.
Instead they are removing the 'r' set dammit!!!!
hahaha

**Comment by kenmcd** [comment]
It is odd that InDesign does not properly support Character Variants (cvNN).
Affinity Publisher and LibreOffice fully support Character Variants.

InDesign does support Access All Alternates (aalt) - which should include all the Character Variants.
They create this feature even if it is not actually coded into the font. A font may or may not already have this OpenType feature. Inter does have aalt.
When you highlight a character in InDesign and you get that little dialog of all the alternates - that is aalt.
I know that Access All Alternates is available in the paragraph styles menu in QuarkXPress - so perhaps InDesign also has this feature in their style settings.

**Comment by acordy** [comment]
Yeah, it is odd that InDesign hasn't implemented support for Character Variants the way they have for Stylistic Sets, not sure if it'll eventually show up or if it's not on their radar. I can't remember exactly where you can file issues or feature requests with them (though I've seen it somewhere before), but I may look into it at some point (wouldn't surprise me if others have already requested this).

The alternates are indeed there as a little context menu when you select a character, but for the default "a" it doesn't display the one-storey "a" that @dpor is looking for. Even if it did, it would have to be selected manually for every single one, which is obviously not ideal. I still think the best bet for anyone looking to use character variants as defaults in InDesign is to freeze them and export new fonts. Not ideal, but the best option I can think of.

**Reply by dpor** [reply]
You can access the alternate 'a' by selecting Stylistic Alternates within Adobe Illustrator which is good.
But Indesign only has Contextual Alternatives which I thought would do the same thing but seems to do nothing.
Thanks again for the input and feedback. 

![image](https://github.com/rsms/inter/assets/3615115/e409e1d0-2db6-4f6a-b5cd-f9788d60ee57)
![image](https://github.com/rsms/inter/assets/3615115/70f91ee7-3ac6-4176-b867-8446a54b8e07)
